main/src/ca/uwaterloo/flix/language/ast/SourcePosition.scala

```diff
@@ -1,43 +1,78 @@
+  def mkFromZeroIndexed(source: Source, line: Int, col: Int): SourcePosition =
+    new SourcePosition(source, lineOneIndexed = line + 1, colOneIndexed = (col + 1).toShort)
 
   /**
-    * [[PartialOrdering]] describing if and how two [[SourcePosition]] are ordered.
+    * Returns a [[SourcePosition]] in `source` with the zero-indexed `line` and `col`.
+    *
+    * `line` and `col` must be 1 or greater.
     */
+  def mkFromOneIndexed(source: Source, line: Int, col: Int): SourcePosition =
+    new SourcePosition(source, lineOneIndexed = line, colOneIndexed = col.toShort)
+
+  /** An unknown source position. */
+  val Unknown: SourcePosition = firstPosition(Source(Input.Unknown, Array.emptyCharArray))
+
+  /**
+    * Returns a new [[SourcePosition]] where the column of `pos` is reduced by 1.
+    *
+    * OBS: Crashes if the column cannot be reduced.
+    */
+  def moveLeft(pos: SourcePosition): SourcePosition = {
+    if (pos.colOneIndexed <= 1) throw InternalCompilerException(s"Trying to reduce the column of ${SourceLocation.zeroPoint(isReal = false, pos).toString}", SourceLocation.Unknown)
+    else new SourcePosition(pos.source, pos.lineOneIndexed, (pos.colOneIndexed - 1).toShort)
+  }
+
+  /**
+    * Returns a new [[SourcePosition]] where the column of `pos` is increased by 1.
+    *
+    * OBS: This new position might not be a real position in the source.
+    */
+  def moveRight(pos: SourcePosition): SourcePosition = {
+    new SourcePosition(pos.source, pos.lineOneIndexed, (pos.colOneIndexed + 1).toShort)
+  }
+
+  /** [[PartialOrdering]] for [[SourcePosition]]s. */
   implicit object PartialOrder extends PartialOrdering[SourcePosition] {
+
     /**
-      * Returns a comparison between `x` and `y` if the two are comparable. Returns [[None]] otherwise.
-      *
-      * If `x` is less than `y`, the returned number is negative,
-      * if `x` is greater than `y`, it's positive and
-      * if they are the same, it's 0.
+      * Returns a comparison (`Some(cmp)`) between `x` and `y` if they point to the same file.
+      * Returns `None` otherwise.
       *
-      * @param x  The [[SourcePosition]] to be compared to `y`
-      * @param y  The [[SourcePosition]] to be comapred to `x`
-      * @return   A comparison between `x` and `y` if they're comparable, [[None]] otherwise.
+      *   - If `x` is strictly before `y` then `cmp < 0`
+      *   - If `x` is strictly after `y` then `cmp > 0`
+      *   - If they are the same then `cmp == 0`.
       */
     override def tryCompare(x: SourcePosition, y: SourcePosition): Option[Int] = {
-      if (x.source != y.source) { return None }
-      if (x.line != y.line) { return Some(x.line - y.line) }
-      if (x.col != y.col) { return Some(x.col - y.col) }
-      Some(0)
+      if (x.source != y.source) None
+      else if (x.lineOneIndexed != y.lineOneIndexed) Some(x.lineOneIndexed - y.lineOneIndexed)
```

Comment:
Another option maybe:

```scala
(x.lineOneIndex, x.colOneIndexed).compare((y.lineOneIndexed, y.colOneIndexed))
```

Reply:
I think i like that better but ill keep this PR to one thing